packages/skill-api/src/core/index.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  SkillRecordingsAction,
+  SkillRecordingsHandlerParams,
+  SkillRecordingsHeader,
+} from './types'
+import {init} from './init'
+import renderPage from './pages'
+import {sendFeedbackFromUser} from './services/send-feedback-from-user'
+
+export interface OutgoingResponse<
+  Body extends string | Record<string, any> | any[] = any,
+> {
+  status?: number
+  headers?: SkillRecordingsHeader[]
+  body?: Body
+  redirect?: string
+  cookies?: any[]
+}
+
+export interface IncomingRequest {
+  /** @default "http://localhost:3000" */
+  host?: string
+  method?: string
+  cookies?: Record<string, string>
+  headers?: Record<string, any>
+  query?: Record<string, any>
+  body?: Record<string, any>
+  action: SkillRecordingsAction
+  providerId?: string
+  error?: string
+}
+
+export async function SkillRecordingsHandler<
+  Body extends string | Record<string, any> | any[],
+>(params: SkillRecordingsHandlerParams): Promise<OutgoingResponse<Body>> {
+  const {options: userOptions, req, token} = params
+
+  // TODO: implement errors
+  const {action, error, method = 'GET'} = req
+
+  const {options, cookies} = await init({
+    userOptions,
+    action,
+    host: req.host,
+    cookies: req.cookies,
+    isPost: method === 'POST',
+  })
+
+  if (method === 'GET') {
+    const render = renderPage({...options, query: req.query, cookies})
+
+    // TODO: implement override pages
+    //  pages are overrides so that you can render user defined pages
+    // but not implemented here
+    const {pages} = options
+
+    switch (action) {
+      case 'test':
+        return render.test()
```

Comment:
in this case we are rendering an **HTML page** and returning that so it will display for the user. Under the hood it uses a lightweight preact compiler to convert JSX to the page.